server/api/order.go

```diff
@@ -0,0 +1,43 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"server/model"
+)
+
+func CreateOrder(w http.ResponseWriter, r *http.Request) {
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "Error reading request body", http.StatusInternalServerError)
+		return
+	}
+
+	var order model.Order
+	err = json.Unmarshal(body, &order)
```

Comment:
I am not sure whether it is necessary. But do we need to double check the roder content before calling Unmarshal?

Reply:
This seems not necessary.